dali/ft/daft.cpp

```diff
@@ -99,6 +99,14 @@ void CDistributedFileSystem::import(IFileDescriptor * from, IDistributedFile * t
     sprayer->setPartFilter(filter);
     sprayer->setSource(from);
     sprayer->setTarget(to);
+
+    bool compressInput = from->isCompressed();
+    bool compressOutput = options->getPropBool("@compress");
+    bool preserveCompression = options->getPropBool("@preserveCompression");
+    LOG(MCdebugInfo, unknownJob, "DFS: compressInput:%d, compressOutput:%d, preserveCompression:%d ", compressInput, compressOutput, preserveCompression);
+    if (preserveCompression && compressInput)
```

Comment:
What if compressOutput = true and preserveCompression = false?

I would expect output to be compressed - but this code doesn't look like it will do that, is compression turned on somewhere else or this is a bug here?

Reply:
It works because the target compression already set in sprayer->setTarget(to);
However it will be more cleaner solution if I merge this preserveCompression check with the existing target compression code in sprayer->setTarget(to);